### DISCUSSION TITLE
Convert the HANDLE to a fltk-rs Image

### DISCUSSION THREAD

**Discussion by minghamn** [fltk-rs/fltk-rs#1331] [Q&A]
How to convert the HANDLE to fltk-rs Image.
I want to load icons from an exe or dll resource and display them.
Thank you !

```
fn get_exe_icon(path: &str) {
    let path: Vec<u16> = OsStr::new(path).encode_wide().chain(Some(0)).collect();
    let name: Vec<u16> = OsStr::new("app_icon")
        .encode_wide()
        .chain(Some(0))
        .collect();
    unsafe {
         // windows::Win32::System::LibraryLoader::LoadLibraryExW
        let hmodule = LoadLibraryExW(
            PCWSTR::from_raw(path.as_ptr()),
            HANDLE::default(),
            LOAD_LIBRARY_AS_DATAFILE,
        )
        .unwrap();

         // windows::Win32::UI::WindowsAndMessaging::LoadImageW
        let himage =
            LoadImageW(hmodule, IDI_APPLICATION, IMAGE_ICON, 0, 0, LR_DEFAULTSIZE).unwrap();

        // Convert the HANDLE to a fltk-rs Image
        // What should I do next to display it in fltk-rs
    }
}
```

**Comment by MoAlyousef** [comment]
If it’s for handling ico files, there’s a new image::IcoImage type which hasn’t been released yet in fltk-rs, or you could use the image crate which supports ico files and can be converted to an fltk image as can be seen here:
https://github.com/fltk-rs/demos/blob/master/image/src/main.rs

Otherwise to use it thru the win32 api, you would need to construct a PICTDESC and call IPicture::SaveAsFile on a win32 stream, get the size of the stream and a pointer to it, then create an fltk RgbImage from this data:
https://docs.rs/fltk/latest/fltk/image/struct.RgbImage.html

Alternatively you would convert the HICON to HBITMAP and use GetDIBits() to get the underlying data.

**Reply by minghamn** [reply]
This is a bit difficult for me, I will try to achieve it, thank you for your answer.
